python/plugins/processing/tests/testdata/expected/grid_diamond.gml

```diff
@@ -1,214 +0,0 @@
-<?xml version="1.0" encoding="utf-8" ?>
```

Comment:
You're right, I have not quite explained my proposal. 
There are some differences between the proposed alg and the original alg. Obviously this is a new option for overlay, but also:
* The point of origin is located at the bottom left and not at the top left as on the original algo; It may be an error.
* The original algo does not fully cover the extent for diamond/rectangle and does not necessarily end the grid for the lines, as user it is a point that I can regret.

That is why I proposed new tests.

However, I have prepared another version more current friendly where the existing tests are respected except for GridLine where I propose to totally grid the extent like GridPolygon.